core/src/main/java/org/geysermc/geyser/item/GeyserCustomItemData.java

```diff
@@ -211,6 +273,80 @@ public Builder tags(@Nullable Set<String> tags) {
             return this;
         }
 
+        @Override
+        public Builder stackSize(int stackSize) {
+            if (stackSize < 1) {
+                throw new IllegalArgumentException("Stack size cannot be below 1 (" + stackSize + " was given)");
+            } else if (stackSize > 1) {
+                if (this.maxDamage > 0) {
+                    throw new IllegalArgumentException("Stack size cannot be above 1 when max damage is above 0 (" + stackSize + " was given)");
+                }
+                // Explicitly set max damage to 0 instead of falling back to the Java vanilla item value
+                this.maxDamage = 0;
+            }
+
+            this.stackSize = stackSize;
+            return this;
+        }
+
+        @Override
+        public Builder maxDamage(int maxDamage) {
+            if (maxDamage < 0) {
+                throw new IllegalArgumentException("Max damage cannot be below 0 (" + maxDamage + " was given)");
+            } else if (maxDamage > 0) {
+                if (this.stackSize > 1) {
+                    throw new IllegalArgumentException("Max damage cannot be above 0 when stack size is above 1 (" + maxDamage + " was given)");
+                }
+                // Explicitly set stack size to 1 instead of falling back to the Java vanilla item value
+                this.stackSize = 1;
+            }
+
+            this.maxDamage = maxDamage;
+            return this;
+        }
+
+        @Override
+        public Builder attackDamage(int attackDamage) {
+            this.attackDamage = attackDamage;
```

Comment:
If applicable, would be nice to annotate and check attackDamage and protectionValue as well